Classes/EventListener/EventListenerLocator.php

```diff
@@ -80,43 +81,34 @@ protected static function detectListeners(ObjectManagerInterface $objectManager)
         $listeners = [];
         /** @var ReflectionService $reflectionService */
         $reflectionService = $objectManager->get(ReflectionService::class);
-        /** @var EventTypeResolver $reflectionService */
-        $eventTypeService = $objectManager->get(EventTypeResolver::class);
+        /** @var EventTypeResolver $eventTypeResolver */
+        $eventTypeResolver = $objectManager->get(EventTypeResolver::class);
         foreach ($reflectionService->getAllImplementationClassNamesForInterface(EventListenerInterface::class) as $listenerClassName) {
             foreach (get_class_methods($listenerClassName) as $methodName) {
-                $eventType = $eventClassName = null;
-
-                preg_match('/^when.*$/', $methodName, $matches);
+                preg_match('/^when[A-Z].*$/', $methodName, $matches);
```

Comment:
Probably not really relevant but now it only matches methods with a correct name